Runtimes/Supplemental/cmake/modules/FindSwiftCore.cmake

```diff
@@ -0,0 +1,123 @@
+#[=======================================================================[.rst:
+FindSwiftCore
+------------
+
+Find SwiftCore, deferring to SwiftCoreConfig.cmake when requested.
+This is meant to find the core library to be linked by the Supplemental libraries.
+
+Imported Targets
+^^^^^^^^^^^^^^^^
+
+The following :prop_tgt:`IMPORTED` TARGETS may be defined:
+
+ ``SwiftCore``
+
+Hint Variables
+^^^^^^^^^^^^^^
+
+ ``SDKROOT``
+   Set the path to the Swift SDK Root.
+   This only affects Windows builds.
+
+ ``Swift_SDKROOT``
+   Set the path to the Swift SDK installation.
+   This only affects Linux and Windows builds.
+   Apple builds always use the library provided by the SDK.
+
+Result Variables
+^^^^^^^^^^^^^^^^
+
+The module may set the following variables if `SwiftCore_DIR` is not set.
+
+ ``SwiftCore_FOUND``
+   true if core was found
+
+ ``SwiftCore_LIBRARIES`` OR ``SwiftCore_IMPLIB``
```

Comment:
I don't think that we should be doing these variables. Instead we should create a `SwiftCore` target that is an imported library (static or shared). On Windows, we can associate an `IMPORTED_IMPLIB` property on the shared library target. This description does seem to match what we are doing below.